### PR TITLE
sim: fix text-segment not taking effect

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
fix text-segment not taking effect

### CODE REVIEW

**Review by anchao — APPROVED** [review]

File: boards/sim/sim/sim/scripts/Make.defs [hunk 1]
```diff
@@ -272,6 +272,6 @@ endif
 # Let the symbol table link from 0x400000
 # which is more convenient for debugging.
 
-ifeq ($(CONFIG_HOST_MACOS),n)
+ifeq ($(CONFIG_HOST_MACOS),)
   LDFLAGS += -Wl,-Ttext-segment=0x400000
```
Comment (pkarashchenko) [inline boards/sim/sim/sim/scripts/Make.defs:276]: Maybe we can merge this with line 262?

**Comment by pkarashchenko** [comment]
Please rebase PR to fix CI failures

File: boards/sim/sim/sim/scripts/Make.defs (hunk 1 above)
Comment (Gary-Hobson) [reply boards/sim/sim/sim/scripts/Make.defs]: done

File: boards/sim/sim/sim/scripts/Make.defs [hunk 2]
```diff
@@ -259,6 +259,9 @@ LDELFFLAGS += -T $(call CONVERT_PATH,$(TOPDIR)/binfmt/libelf/gnu-elf.ld)
 ifeq ($(CONFIG_HOST_MACOS),y)
   LDFLAGS += -Wl,-dead_strip
 else
+  # Let the symbol table link from 0x400000
+  # which is more convenient for debugging.
+  LDFLAGS += -Wl,-Ttext-segment=0x400000
   LDFLAGS += -Wl,--gc-sections
```
Comment (pkarashchenko) [inline boards/sim/sim/sim/scripts/Make.defs:265]:
```suggestion
  LDFLAGS += -Wl,--gc-sections
  
  # Let the symbol table link from 0x400000
  # which is more convenient for debugging.
  LDFLAGS += -Wl,-Ttext-segment=0x400000
```

Just to have better visibility that `LDFLAGS += -Wl,-dead_strip` is same as `LDFLAGS += -Wl,--gc-sections` and comment section is not related to `LDFLAGS += -Wl,--gc-sections`

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**